### ISSUE TITLE
SC2215: Comment obscures line continuation

### ISSUE THREAD

**Issue by ednl** [koalaman/shellcheck#1971]
- SC2215 
- version 0.5.0 from repo on WSL 
- [x] The rule's wiki page does not already cover this https://shellcheck.net/wiki/SC2215
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/usr/bin/env bash
find . \                # look in dir
        -name '*.txt'   # for this extension

```

#### Here's what shellcheck currently says:

```
Line 3:
        -name '*.txt'   # for this extension
        ^-- SC2215: This flag is used as a command name. Bad line break or missing [ .. ]?
```

#### Here's what I wanted or expected to see:

I break down long command lines to document individual options. Bash ignores the comment and correctly continues the line from the backslash. Shellcheck should, too.

**Comment by SimonHeimberg** [comment]
This seems to depend on bash version. Bash 4.4.12 does not support # and line continuation.

**Comment by koalaman** [comment]
ShellCheck is right. Bash does not allow `#` after a `\`. 

Your command is equivalent to `find . " "` which finds all files in the current dir plus an error message, then a second error message `line 3: -name: command not found`

**Comment by ednl** [comment]
I am so sorry for polluting your issues. You (and shellcheck) are correct. It does not work in any bash version I have nor can I find the pattern in my scripts. I don't know what I was thinking; perhaps I came from another language, probably one that does not have a line continuation character. Sorry again.
